packages/app/src/app/pages/Pro/pages/InlineCheckout.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { Stack, Text } from '@codesandbox/components';
+import css from '@styled-system/css';
+import { useOvermind } from 'app/overmind';
+import { useScript } from 'app/hooks';
+import { PADDLE_VENDOR_ID } from '../plans';
+
+export const InlineCheckout: React.FC = () => {
+  const {
+    state: {
+      user,
+      activeTeam,
+      pro: { seats, selectedPlan, summary },
+    },
+    actions,
+  } = useOvermind();
+
+  const [scriptLoaded] = useScript('https://cdn.paddle.com/paddle/paddle.js');
+
+  React.useEffect(() => {
+    if (!scriptLoaded) return;
+
+    // @ts-ignore 3rd party integration with global
+    const Paddle = window.Paddle;
+
+    Paddle.Environment.set('sandbox');
+    Paddle.Setup({
+      vendor: PADDLE_VENDOR_ID,
+      eventCallback: event => {
+        if (event.event === 'Checkout.Location.Submit') {
+          const customSummary = event.eventData.checkout.prices.customer;
+          actions.pro.updateSummary({
+            unitPrice: customSummary.unit_price,
+            unit: customSummary.unit,
+            unitTax: customSummary.unit_tax,
+            totalTax: customSummary.total_tax,
+            currency: customSummary.currency,
+            total: customSummary.total,
+          });
+        }
+      },
+    });
+
+    // @ts-ignore 3rd party integration with global
+    window.loadCallback = () => actions.pro.paddleInitialised();
+
+    Paddle.Checkout.open({
+      method: 'inline',
+      product: selectedPlan.id, // Replace with your Product or Plan ID
+      quantity: seats,
+      email: user.email,
+      frameTarget: 'checkout-container', // The className of your checkout <div>
+      loadCallback: 'loadCallback',
+      success: '/pro/success',
+      passthrough: JSON.stringify({ team_id: activeTeam, user_id: user.id }),
+      allowQuantity: true,
+      disableLogout: true,
+      frameInitialHeight: 416,
+      frameStyle: `
+        width: 500px;
+        min-width: 500px;
+        background-color:
+        transparent; border: none;
+      `,
+    });
+  }, [scriptLoaded, seats, user.email, selectedPlan]);
```

Comment:
This could be moved to a hook as `usePaddle`, and that way we can hide ALL the shame there in a hook so it looks cool lol

Reply:
This is the shame file 😅  <InlineCheckout/> is the same as usePaddle()